src/App.tsx

```diff
@@ -1,14 +1,60 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+export function getFilteredTodo(
+  todos: Todo[],
+  filterType: string,
+  query: string,
+) {
+  const filterByType = todos.filter((todo) => {
+    switch (filterType) {
+      case 'active':
+        return todo.completed === false;
+
+      case 'completed':
+        return todo.completed === true;
+
+      default:
+        return todo;
+    }
+  });
+
+  return filterByType.filter(({ title }) => (
+    title.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+  ));
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodo] = useState<Todo[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [selectedTodo, setSelectedTodo] = useState(0);
```

Comment:
You could save only `todoId` in state and then get from there all data that you need.
1 state - instead of 2.